src/PrestaShopBundle/Resources/config/services/bundle/routing.yml

```diff
@@ -6,6 +6,18 @@ services:
         tags: [routing.loader]
 
     prestashop.bundle.routing.legacy_url_converter:
-        class: 'PrestaShopBundle\Routing\LegacyUrlConverter'
+        class: 'PrestaShopBundle\Routing\Converter\LegacyUrlConverter'
         arguments:
             $router: '@router'
+            $legacyRouteProvider: '@=parameter("kernel.environment") == "prod" ? service("prestashop.bundle.routing.cache_provider") : service("prestashop.bundle.routing.router_provider")'
+
+    prestashop.bundle.routing.router_provider:
+        class: 'PrestaShopBundle\Routing\Converter\RouterProvider'
+        arguments:
+            $router: '@router'
```

Comment:
For the reccord, since when we use named arguments like $router? /c @sarjon.

Please stop, this is inconsistent with the existing service declaration files. We will update all the files (and remove most of them) in the next major version.

Reply:
Well both declaration can work and cohabit fine together, so as you said it is only a matter of consistency (coding style)
Although I think this notation is less close to modifications, you can change a constructor and insert a new parameter without breaking the service definition
And it also helps to ready the service.yml file as you can see the argument name which would help outside developers to override/reuse our core services

Reply:
the problem i see, is that once you change argument name in constructor and you must remember to update it in service definition. :/

Reply:
true! nothing is perfect in all situations of course, but from my experience it happens more often to add arguments to a constructor than to change its arguments names
anyway, I feel like I am loosing on this question ^^ any other opinions? :P
@matks @PierreRambaud @eternoendless ? ^^

Reply:
Both of you are rights. One reason I could prefer what you write @jolelievre is because of:
https://symfony.com/blog/new-in-symfony-3-4-local-service-binding

Reply:
we won't use autowiring in 1.7, that's the point: this will make the learning of 1.7 even harder as we must keep the already declared services. Remember: bad but consistent > good but inconsistent.

Reply:
ok I'm gonna change the notation